potodo/_po_file.py

```diff
@@ -53,15 +53,28 @@ def __lt__(self, other: "PoFileStats") -> bool:
         return self.filename < other.filename
 
 
-def get_po_files_from_repo(repo_path: str) -> Mapping[str, Sequence[PoFileStats]]:
-    """Gets all the po files from a given repository.  Will return a list
-    with all directories and PoFile instances of `.po` files in those
-    directories.
+def is_within(file: Path, folder: Path) -> bool:
+    """ Check if `file` is within `folder`'s tree.
     """
+    folder = folder.resolve()
+    file = file.resolve()
+    return any(parent == folder for parent in file.parents)
```

Comment:
```suggestion
    return folder in file.parents
```

Reply:
Beware, in our implementations, we cannot use `--exclude sphinx.po`, only exclude directories.

`folder in file.parents or folder == file` would allow excluding files, which I think is nice (you'll have to change the `help` of `--exclude` if you go that way).

Reply:
`return any(parent == folder for parent in file.parents)`
Dafuq did I wrote?

> Beware, in our implementations, we cannot use `--exclude sphinx.po`, only exclude directories.

Yes. I aimed for simplicity (end goal is to exclude whatsnew and c-api etc. folders into doc).
 
> `folder in file.parents or folder == file` would allow excluding files, which I think is nice (you'll have to change the `help` of `--exclude` if you go that way).

Of course. Will make necessary changes if you think excluding files bears value.